### PR TITLE
feat: Save landscape polygon area on landscape upate (#210)

### LINKED ISSUE #210: Landscape Profile: Add Landscape size field
## Description
- Add landscape size field

## Additional context

Figma: https://www.figma.com/file/HsAlxOnHkABJ3IyT2D0SRW/Create%2FUpdate-Landscape-Profile_V2-R1.2

### PR DESCRIPTION
**Pull request by david-code**
## Description

Calculates the area of a landscape boundary polygon. Saves the value to the database, and updates it whenever the boundary is updated. Returns this value in the frontend API call.

### Checklist
- [x] Corresponding issue has been opened
- [x] New tests added

### Related Issues
Fixes #210

### Verification steps

TODO

### CODE REVIEW

File: terraso_backend/apps/core/models/landscapes.py [hunk 1]
```diff
@@ -49,6 +51,9 @@ class Meta(SlugModel.Meta):
         }
 
     def save(self, *args, **kwargs):
+        if self.area_polygon:
+            self.area_scalar_m2 = calculate_geojson_feature_area(self.area_polygon)
```
Comment (josebui) [inline terraso_backend/apps/core/models/landscapes.py:55]: Question: @david-code Is this what you were talking about in the slack thread about dirty fields? It seems that right now every time the landscape data is updated the area_scalar field will be recalculated, right? so this will be triggered even if the area polygon value is the same, right?

**Review by josebui — APPROVED** [review]
Looks good, my only concern is caluculating this value everytime landscape is saved, how expensive is the process?

File: terraso_backend/apps/core/models/landscapes.py (hunk 1 above)
Comment (david-code) [reply terraso_backend/apps/core/models/landscapes.py]: Yes, exactly. Looking a bit more at this, I think I will try installing the django-dirty-fields library. It seems like the best solution to me.

**Comment by david-code** [comment]
Added the `django-dirtyfields` library and rebased.

File: terraso_backend/apps/core/geo.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+"""Geospatial utility methods"""
+from pyproj import CRS
+from shapely.geometry import shape
+
+# CRS that should be used for all GeoJSON (see https://www.rfc-editor.org/rfc/rfc7946#section-4)
+DEFAULT_CRS = CRS.from_string("urn:ogc:def:crs:OGC:1.3:CRS84")
```
Comment (paulschreiber) [inline terraso_backend/apps/core/geo.py:6]: question: Should this be in settings.py?

File: terraso_backend/apps/core/geo.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+
+def calculate_geojson_feature_area(feature_json):
+    try:
+        features = feature_json["features"]
+        if len(features) != 1:
+            raise ValueError(f"Expecting only 1 feature, but saw {len(features)}")
```
Comment (paulschreiber) [inline terraso_backend/apps/core/geo.py:13]: suggestion: add a comment explaining why we only support 1 feature.

File: terraso_backend/apps/core/geo.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    """Calculates the area of a polygon supplied in GeoJSON format. Default CRS is WSG84 if
+    none are provided"""
+    untransformed = shape(polygon_json)
+    geod = DEFAULT_CRS.get_geod()
+    area, _perimeter = geod.geometry_area_perimeter(untransformed)
+    return abs(area)
+
+
+def m2_to_hectares(area):
+    return area / 1000
```
Comment (paulschreiber) [inline terraso_backend/apps/core/geo.py:35]: comment: this is fine for now, but if we do a lot of converting, we should use a library, i.e. https://pypi.org/project/area-unit-conversion/

File: terraso_backend/apps/core/migrations/0024_landscape_profile_image.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
+# Generated by Django 4.1 on 2022-11-07 15:24
```
Comment (paulschreiber) [inline terraso_backend/apps/core/migrations/0024_landscape_profile_image.py:1]: issue: this migration looks unrelated to area calculations and probably belongs in another branch.

File: terraso_backend/apps/core/models/landscapes.py [hunk 6]
```diff
@@ -62,6 +65,8 @@ class Landscape(SlugModel):
         choices=MEMBERSHIP_TYPES,
         default=DEFAULT_PARTNERSHIP_STATUS,
     )
+    profile_image = models.URLField(blank=True, default="")
+    profile_image_description = models.TextField(blank=True, default="")
```
Comment (paulschreiber) [inline terraso_backend/apps/core/models/landscapes.py:69]: issue: looks like these are from another branch.

File: terraso_backend/apps/graphql/schema/landscapes.py [hunk 7]
```diff
@@ -40,16 +41,26 @@ class Meta:
...
         )
+
         interfaces = (relay.Node,)
         connection_class = TerrasoConnection
 
+    area_scalar_ha = graphene.Float()
+
+    def resolve_area_scalar_ha(self, info):
+        area = self.area_scalar_m2
+        return round(m2_to_hectares(area), 3) if area is not None else None
```
Comment (paulschreiber) [inline terraso_backend/apps/graphql/schema/landscapes.py:62]: question: would this be easier to read as?
```
        return None if area is None else round(m2_to_hectares(area), 3) 
```

File: terraso_backend/apps/storage/forms.py [hunk 8]
```diff
@@ -0,0 +1,35 @@
+import structlog
```
Comment (paulschreiber) [inline terraso_backend/apps/storage/forms.py:1]: issue: looks to be form the wrong branch

File: terraso_backend/apps/storage/services.py [hunk 9]
```diff
@@ -71,3 +71,6 @@ class ProfileImageService(UploadService):
 
     def get_path_on_storage(self, user_id, file_name):
         return f"{user_id}/profile-image"
+
```
Comment (paulschreiber) [inline terraso_backend/apps/storage/services.py:74]: issue: looks to be from the wrong branch

File: terraso_backend/apps/storage/urls.py [hunk 10]
```diff
@@ -1,8 +1,17 @@
 from django.urls import path
 from django.views.decorators.csrf import csrf_exempt
 
-from apps.storage.views import ProfileImageView
+from apps.storage.views import LandscapeProfileImageView, UserProfileImageView
```
Comment (paulschreiber) [inline terraso_backend/apps/storage/urls.py:4]: issue: looks to be from the wrong branch

File: terraso_backend/tests/storage/test_views.py [hunk 11]
```diff
@@ -1,7 +1,9 @@
 from unittest import mock
+from unittest.mock import patch
```
Comment (paulschreiber) [inline terraso_backend/tests/storage/test_views.py:2]: issue: looks to be from the wrong branch

File: terraso_backend/apps/storage/views.py [hunk 12]
```diff
@@ -1,15 +1,20 @@
+from dataclasses import asdict
```
Comment (paulschreiber) [inline terraso_backend/apps/storage/views.py:1]: issue: looks to be from the wrong branch

**Review by paulschreiber — CHANGES_REQUESTED** [review]

File: terraso_backend/apps/core/migrations/0024_landscape_profile_image.py (hunk 5 above)
Comment (david-code) [reply terraso_backend/apps/core/migrations/0024_landscape_profile_image.py:1]: Yes, will figure out why these here and get rid of those commits.

**Review by paulschreiber — APPROVED** [review]
